sdk/typescript/src/client/cryptography.ts

```diff
@@ -421,11 +470,47 @@ export async function verifySecpSignature(
 /**
  * Create a public key from a DWallet output.
  *
+ * @param curve - The curve to use for key generation
  * @param dWalletOutput - The DWallet output
  * @returns The public key
  */
-export async function publicKeyFromDWalletOutput(dWalletOutput: Uint8Array): Promise<Uint8Array> {
-	return Uint8Array.from(await public_key_from_dwallet_output(dWalletOutput));
+export async function publicKeyFromDWalletOutput(
+	curve: Curve,
+	dWalletOutput: Uint8Array,
+): Promise<Uint8Array> {
+	const raw = await public_key_from_dwallet_output(curve, dWalletOutput);
+	const encoded = new Uint8Array(raw); // Convert JS array -> Uint8Array
+
+	// If it is Ristretto or ED25519
+	// It is not BCS-Encoded since they have fixed size in rust
```

Comment:
I don't get this function. All options including these ones are bcs encoded. What is the output type you are trying to get anyways? you should deserialize in Rust/wasm, and then encode it in whatever format you wish as bytes in Rust, and here you shouldn't do any logic @Iamknownasfesal